File: src/com/adventofcode/december15/MainProgram.java
package com.adventofcode.december15;

public class MainProgram {

    public static void main(String[] args) {
        new MainProgram();
    }

    //-----------------------------------------------------------

    private CavernMap cavernMap;

    private static final boolean test = false;

    public MainProgram() {
        String fileName = getFileName(test);
        Input input = new Input(fileName);
        cavernMap = input.getBaseClass(true);

        long output = run();
        System.out.println("Puzzle output : " + output);
    }

    public long run() {
//        cavernMap.print();
        long sum = cavernMap.findShortestPath();
        System.out.println("Nodes visited " + Position.visitCount);
        System.out.println("Nodes updated " + Position.updateCount);
        return sum;
    }

    private String getFileName(boolean test) {
        int day = getDayOfMonthFromClassName();
        return "input" + String.format("%02d", day) + "_" + (test ? "example" : "1");
    }

    private int getDayOfMonthFromClassName() {
        String monthName = "december";
        int i = this.getClass().getName().indexOf(monthName);
        int j = this.getClass().getName().lastIndexOf(".");
        String dayOfMonth = this.getClass().getName().substring(i + monthName.length(), j);
        return Integer.parseInt(dayOfMonth);
    }

}

File: src/com/adventofcode/december02/MainProgram.kt
package com.adventofcode.december02

fun main() {
    MainProgram().MainProgram()
}

class MainProgram {
    private val test = false

    fun MainProgram() {
        val fileName = getFileName(test)
        val input = Input(fileName)
        val baseClass = BaseClass(input)
//        val output = baseClass.totalScore()
        val output = baseClass.totalScore2()
        println("Puzzle output : $output")
    }

    //------------------------------------------------------------------------------------------------------------------

    private fun getFileName(test: Boolean): String {
        val day = getDayOfMonthFromClassName()
        return "input" + String.format("%02d", day) + "_" + if (test) "example" else "1"
    }

    private fun getDayOfMonthFromClassName(): Int {
        val monthName = "december"
        val i = this.javaClass.name.indexOf(monthName)
        val j = this.javaClass.name.lastIndexOf(".")
        val dayOfMonth = this.javaClass.name.substring(i + monthName.length, j)
        return dayOfMonth.toInt()
    }
}